xarray/backends/netcdf3.py

```diff
@@ -112,7 +112,21 @@ def _maybe_prepare_times(var):
     if data.dtype.kind in "iu":
         units = var.attrs.get("units", None)
         if units is not None and coding.variables._is_time_like(units):
-            mask = data == np.iinfo(np.int64).min
+            default_int64_fill_value = np.iinfo(np.int64).min
+            default_int32_fill_value = np.iinfo(np.int32).min
+            mask = data == default_int64_fill_value
+
+            if var.attrs.get("_FillValue") == default_int64_fill_value:
+                if (data == default_int32_fill_value).any():
+                    raise ValueError(
+                        f"Could not safely coerce default int64 _FillValue "
+                        f"({default_int64_fill_value}) to the analogous int32 "
+                        f"value ({default_int32_fill_value}), since it "
+                        f"already exists as non-missing within variable "
+                        f"{name!r}. Try explicitly setting "
+                        f"encoding['_FillValue'] to another int32 value."
+                    )
```

Comment:
I don't think we need this error. We already have a check in `coerce_nc3_dtype` that the encoded int32 values match the original int64 values exactly.

Reply:
I think this is needed to check something different, i.e. that the new fill value does not conflict with an existing value in the array.  I believe the downstream check in `coerce_nc3_dtype` would pass in that circumstance since we replace the int64 fill value with the new int32 fill value upstream of it.